### PR TITLE
Add support phan_client for php

### PR DESCRIPTION
**Pull request by haginaga**
I've added support phan_client for php. (#1561)

In .vimrc  file to enable to use the phan_client, 
```
let g:ale_php_phan_use_client = 1
let g:ale_php_phan_executable = 'phan_client'
```

### CODE REVIEW

File: ale_linters/php/phan.vim [hunk 1]
```diff
@@ -1,36 +1,71 @@
...
+    endif
+
+    return l:executable
+endfunction
+
 function! ale_linters#php#phan#GetCommand(buffer) abort
-    return 'phan -y '
-    \   . ale#Var(a:buffer, 'php_phan_minimum_severity')
-    \   . ' %s'
+    if g:ale_php_phan_use_client == 1
```
Comment (w0rp) [inline ale_linters/php/phan.vim]: Use `ale#Var` for this variable too.

File: ale_linters/php/phan.vim [hunk 2]
```diff
@@ -1,36 +1,71 @@
...
 
 " Define the minimum severity
 let g:ale_php_phan_minimum_severity = get(g:, 'ale_php_phan_minimum_severity', 0)
 
+let g:ale_php_phan_executable = get(g:, 'ale_php_phan_executable', 'phan')
+let g:ale_php_phan_use_client = get(g:, 'ale_php_phan_use_client', 0)
+
+function! ale_linters#php#phan#GetExecutable(buffer) abort
+    let l:executable = ale#Var(a:buffer, 'php_phan_executable')
+    if g:ale_php_phan_use_client == 1 && l:executable is# 'phan'
```
Comment (w0rp) [inline ale_linters/php/phan.vim]: Put a new line before if statements that don't start a new level of indentation.

File: ale_linters/php/phan.vim [hunk 3]
```diff
@@ -1,36 +1,71 @@
...
+    if g:ale_php_phan_use_client == 1
+        " Phan error: ERRORTYPE: message in /path/to/some-filename.php on line nnn
+        let l:pattern = '^Phan error: \(\w\+\): \(.\+\) in \(.\+\) on line \(\d\+\)$'
+    else
+        " /path/to/some-filename.php:18 ERRORTYPE message
+        let l:pattern = '^.*:\(\d\+\)\s\(\w\+\)\s\(.\+\)$'
+    endif
 
+    let l:output = []
     for l:match in ale#util#GetMatches(a:lines, l:pattern)
```
Comment (w0rp) [inline ale_linters/php/phan.vim]: Put a new line before `for`, where it doesn't start a new level of indentation.

File: ale_linters/php/phan.vim [hunk 4]
```diff
@@ -1,36 +1,71 @@
...
+    return ale#Escape(l:executable) . ' ' . l:args
 endfunction
 
 function! ale_linters#php#phan#Handle(buffer, lines) abort
     " Matches against lines like the following:
-    "
-    " /path/to/some-filename.php:18 ERRORTYPE message
-    let l:pattern = '^.*:\(\d\+\)\s\(\w\+\)\s\(.\+\)$'
-    let l:output = []
+    if g:ale_php_phan_use_client == 1
```
Comment (w0rp) [inline ale_linters/php/phan.vim]: Use `ale#Var` here too.

**Comment by haginaga** [comment]
@w0rp Thank you for reviewing my PR.  I fixed. Please re-review by you.

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**